File: 2005/Windows SDK NetFx 3.0/Technologies/ComponentServices/Transactions/CS/Transactions/TxForm.cs
/*=====================================================================
  File:      TXForm.cs

  Summary:   Windows.Forms Code for .NET COM+ Transactions Sample

---------------------------------------------------------------------
  This file is part of the Microsoft .NET Framework SDK Code Samples.

  Copyright (C) Microsoft Corporation.  All rights reserved.

This source code is intended only as a supplement to Microsoft
Development Tools and/or on-line documentation.  See these other
materials for detailed information regarding Microsoft code samples.

THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
PARTICULAR PURPOSE.
=====================================================================*/

using System;
using System.Drawing;
using System.Collections;
using System.Globalization;
using System.ComponentModel;
using System.Windows.Forms;

namespace Microsoft.Samples.Technologies.ComponentServices.Transactions
{
	/// <summary>
	/// This class handles the UI for the Transaction Sample
	/// </summary>
	public class TXForm : System.Windows.Forms.Form
	{
		private System.Windows.Forms.Label currentValueLabel;
		private System.Windows.Forms.TextBox currentValue;
		private System.Windows.Forms.Button post;
		private System.Windows.Forms.Button autoCompletePost;
		private System.Windows.Forms.TextBox newValue;
		private System.Windows.Forms.Label newValueLabel;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;
		
		public TXForm()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();
		
		}

		protected override void OnLoad(EventArgs e)
		{					
					try
					{
						// The 'using' construct below results in a call to Dispose on exiting the 
						// curly braces. It could be replaced with an explicit call to Object.Dispose
						// This is a C#-specific construct.
						//
						// It is important to dispose COM+ objects as soon as possible so that
						// COM+ metadata such as context does not remain in memory unnecessarily
						// and so that COM+ services such as Object Pooling work properly.
					    using( TXDemoServer txDemoServer = new TXDemoServer()) 
				           //we want the object to be disposed immediately
					   {
						// Retrieve & display the current value
						currentValue.Text = txDemoServer.RetrieveCurrentValue().ToString(CultureInfo.CurrentCulture);
					   }
					}
					catch(CurrentValueNotReadException)
					{
						MessageBox.Show("Unable to read the current value from the database" 
							, "Error");
						Application.Exit();
					}
					catch(Exception ex)
					{
						MessageBox.Show("An exception was caught : "+ex.Message + 
							"\nUnable to launch the COM+ Server. Closing the application." , 
								"Error");
						Application.Exit();
					}
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
            this.currentValueLabel = new System.Windows.Forms.Label();
            this.currentValue = new System.Windows.Forms.TextBox();
            this.post = new System.Windows.Forms.Button();
            this.autoCompletePost = new System.Windows.Forms.Button();
            this.newValue = new System.Windows.Forms.TextBox();
            this.newValueLabel = new System.Windows.Forms.Label();
            this.SuspendLayout();
// 
// currentValueLabel
// 
            this.currentValueLabel.Location = new System.Drawing.Point(8, 8);
            this.currentValueLabel.Name = "currentValueLabel";
            this.currentValueLabel.Size = new System.Drawing.Size(80, 16);
            this.currentValueLabel.TabIndex = 0;
            this.currentValueLabel.Text = "Current Value:";
// 
// currentValue
// 
            this.currentValue.Location = new System.Drawing.Point(88, 8);
            this.currentValue.Name = "currentValue";
            this.currentValue.ReadOnly = true;
            this.currentValue.TabIndex = 1;
            this.currentValue.TabStop = false;
// 
// post
// 
            this.post.Location = new System.Drawing.Point(8, 72);
            this.post.Name = "post";
            this.post.Size = new System.Drawing.Size(88, 23);
            this.post.TabIndex = 3;
            this.post.Text = "&Post";
            this.post.Click += new System.EventHandler(this.Post_Click);
// 
// autoCompletePost
// 
            this.autoCompletePost.Location = new System.Drawing.Point(104, 72);
            this.autoCompletePost.Name = "autoCompletePost";
            this.autoCompletePost.Size = new System.Drawing.Size(88, 23);
            this.autoCompletePost.TabIndex = 4;
            this.autoCompletePost.Text = "&AutoComplete";
            this.autoCompletePost.Click += new System.EventHandler(this.AutoCompletePost_Click);
// 
// newValue
// 
            this.newValue.Location = new System.Drawing.Point(88, 40);
            this.newValue.Name = "newValue";
            this.newValue.TabIndex = 2;
            this.newValue.Text = "5";
// 
// newValueLabel
// 
            this.newValueLabel.Location = new System.Drawing.Point(8, 40);
            this.newValueLabel.Name = "newValueLabel";
            this.newValueLabel.Size = new System.Drawing.Size(80, 16);
            this.newValueLabel.TabIndex = 5;
            this.newValueLabel.Text = "New Value:";
// 
// TXForm
// 
			this.AutoScaleDimensions = new System.Drawing.SizeF(5, 13);
            this.ClientSize = new System.Drawing.Size(200, 101);
            this.Controls.Add(this.newValue);
            this.Controls.Add(this.newValueLabel);
            this.Controls.Add(this.autoCompletePost);
            this.Controls.Add(this.post);
            this.Controls.Add(this.currentValue);
            this.Controls.Add(this.currentValueLabel);
            this.Name = "TXForm";
            this.Text = "TXForm";
            this.ResumeLayout(false);
            this.PerformLayout();

        }
		#endregion

		// Start the Post with a manual transaction if the user clicks on the 
		// Post button
		private void Post_Click(object sender, System.EventArgs e)
		{
			int val = 0;
			try
			{
				val = int.Parse(newValue.Text, CultureInfo.CurrentCulture);
			}
			catch
			{
				MessageBox.Show("Please enter a value to post!");
				return;
			}
			try
			{
				// The 'using' construct below results in a call to Dispose on exiting the 
				// curly braces. It could be replaced with an explicit call to Object.Dispose
				// This is a C#-specific construct.
				//
				// It is important to dispose COM+ objects as soon as possible so that
				// COM+ metadata such as context does not remain in memory unnecessarily
				// and so that COM+ services such as Object Pooling work properly.
			    using( TXDemoServer txDemoServer = new TXDemoServer())
			    //we want the object to be disposed immediately
			   {
		    		txDemoServer.Post(val);
			   }
			}
			catch(DatabaseExecutionException)
			{
				MessageBox.Show("Unable to update the database" , "Error");
				return;
			}
			catch(Exception ex)
			{
				MessageBox.Show("An exception was caught : "+ex.Message);
				return;
			}
			try
			{
				// The 'using' construct below results in a call to Dispose on exiting the 
				// curly braces. It could be replaced with an explicit call to Object.Dispose
				// This is a C#-specific construct.
				//
				// It is important to dispose COM+ objects as soon as possible so that
				// COM+ metadata such as context does not remain in memory unnecessarily
				// and so that COM+ services such as Object Pooling work properly.
				using( TXDemoServer txDemoServer = new TXDemoServer())
				//we want the object to be disposed immediately
				{
					currentValue.Text = txDemoServer.RetrieveCurrentValue().ToString(CultureInfo.CurrentCulture);
				}
			}
			catch(CurrentValueNotReadException)
			{
				MessageBox.Show("Unable to read the current value from the database" ,
										"Error");
				return;
			}
		}

		//Start the Post with automatic transaction if the user clicks on the 
		//autoCompletePost button
		private void AutoCompletePost_Click(object sender, System.EventArgs e)
		{
			int val = 0;
			try
			{
				val = int.Parse(newValue.Text, CultureInfo.CurrentCulture);
			}
			catch
			{
				MessageBox.Show("Please enter a value to post!");
				return;
			}
			try
			{
				// The 'using' construct below results in a call to Dispose on exiting the 
				// curly braces. It could be replaced with an explicit call to Object.Dispose
				// This is a C#-specific construct.
				//
				// It is important to dispose COM+ objects as soon as possible so that
				// COM+ metadata such as context does not remain in memory unnecessarily
				// and so that COM+ services such as Object Pooling work properly.
				using( TXDemoServer txDemoServer = new TXDemoServer())
				//we want the object to be disposed immediately
				{
					txDemoServer.AutoCompletePost(val);
				}
			}
			catch(ValueOutOfRangeException)
			{
				MessageBox.Show("The transaction has been aborted by throwing an " +
											"AbortTransactionException" , "No Error");
			}
			catch(DatabaseExecutionException)
			{
				MessageBox.Show("Unable to update the database" , "Error");
				return;
			}
			catch(Exception ex)
			{
						MessageBox.Show("An exception was caught : " + ex.Message);
						return;
			}
			try
			{
				// The 'using' construct below results in a call to Dispose on exiting the 
				// curly braces. It could be replaced with an explicit call to Object.Dispose
				// This is a C#-specific construct.
				//
				// It is important to dispose COM+ objects as soon as possible so that
				// COM+ metadata such as context does not remain in memory unnecessarily
				// and so that COM+ services such as Object Pooling work properly.
				using( TXDemoServer txDemoServer = new TXDemoServer())
				//we want the object to be disposed immediately
				{
					currentValue.Text = txDemoServer.RetrieveCurrentValue().ToString(CultureInfo.CurrentCulture);
				}
			}
			catch(CurrentValueNotReadException)
			{
				MessageBox.Show("Unable to read the current value from the database" ,
										"Error");
				return;
			}
		}

        [STAThread]
		private static void Main(String[] args)
		{
			Application.Run(new TXForm());
		}
	}
}

File: 2005/Windows SDK NetFx 3.0/Technologies/ComponentServices/Transactions/VB/Transactions/TxForm.vb
'=====================================================================
'  File:      TXForm.vb
'
'  Summary:   Windows.Forms Code for .NET COM+ Transactions Sample
'
'---------------------------------------------------------------------
'  This file is part of the Microsoft .NET Framework SDK Code Samples.
'
'  Copyright (C) Microsoft Corporation.  All rights reserved.
'
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
'
'THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'=====================================================================
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.Globalization
Imports System.ComponentModel
Imports System.Windows.Forms


Namespace Microsoft.Samples.Technologies.ComponentServices.Transactions
    _
    'This class handles the UI for the Transaction Sample
    Public Class TXForm
        Inherits System.Windows.Forms.Form


        Friend WithEvents currentValueLabel As System.Windows.Forms.Label
#Region " Windows Form Designer generated code "
        Friend WithEvents currentValue As System.Windows.Forms.TextBox
        Friend WithEvents post As System.Windows.Forms.Button
        Friend WithEvents autoCompletePost As System.Windows.Forms.Button
        Friend WithEvents newValue As System.Windows.Forms.TextBox
        Friend WithEvents newValueLabel As System.Windows.Forms.Label

        Public Sub New()
            'This call is required by the Windows Form Designer.
            InitializeComponent()

            'Add any initialization after the InitializeComponent() call

        End Sub 'New


        'Form overrides dispose to clean up the component list.
        Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If Not (components Is Nothing) Then
                    components.Dispose()
                End If
            End If
            MyBase.Dispose(disposing)
        End Sub

        'Required by the Windows Form Designer
        Private components As System.ComponentModel.Container

        'NOTE: The following procedure is required by the Windows Form Designer
        'It can be modified using the Windows Form Designer.  
        'Do not modify it using the code editor.
        <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
            Me.currentValueLabel = New System.Windows.Forms.Label
            Me.currentValue = New System.Windows.Forms.TextBox
            Me.post = New System.Windows.Forms.Button
            Me.autoCompletePost = New System.Windows.Forms.Button
            Me.newValue = New System.Windows.Forms.TextBox
            Me.newValueLabel = New System.Windows.Forms.Label
            Me.SuspendLayout()
            '
            'currentValueLabel
            '
            Me.currentValueLabel.Location = New System.Drawing.Point(8, 8)
            Me.currentValueLabel.Name = "currentValueLabel"
            Me.currentValueLabel.Size = New System.Drawing.Size(80, 16)
            Me.currentValueLabel.TabIndex = 0
            Me.currentValueLabel.Text = "Current Value:"
            '
            'currentValue
            '
            Me.currentValue.Location = New System.Drawing.Point(88, 8)
            Me.currentValue.Name = "currentValue"
            Me.currentValue.ReadOnly = True
            Me.currentValue.TabIndex = 1
            Me.currentValue.TabStop = False
            '
            'post
            '
            Me.post.Location = New System.Drawing.Point(8, 72)
            Me.post.Name = "post"
            Me.post.Size = New System.Drawing.Size(88, 23)
            Me.post.TabIndex = 3
            Me.post.Text = "&Post"
            '
            'autoCompletePost
            '
            Me.autoCompletePost.Location = New System.Drawing.Point(104, 72)
            Me.autoCompletePost.Name = "autoCompletePost"
            Me.autoCompletePost.Size = New System.Drawing.Size(88, 23)
            Me.autoCompletePost.TabIndex = 4
            Me.autoCompletePost.Text = "&AutoComplete"
            '
            'newValue
            '
            Me.newValue.Location = New System.Drawing.Point(88, 40)
            Me.newValue.Name = "newValue"
            Me.newValue.TabIndex = 2
            Me.newValue.Text = "5"
            '
            'newValueLabel
            '
            Me.newValueLabel.Location = New System.Drawing.Point(8, 40)
            Me.newValueLabel.Name = "newValueLabel"
            Me.newValueLabel.Size = New System.Drawing.Size(80, 16)
            Me.newValueLabel.TabIndex = 5
            Me.newValueLabel.Text = "New Value:"
            '
            'TXForm
            '
            Me.AutoScaleDimensions = New System.Drawing.SizeF(5, 13)
            Me.ClientSize = New System.Drawing.Size(200, 101)
            Me.Controls.Add(Me.newValue)
            Me.Controls.Add(Me.newValueLabel)
            Me.Controls.Add(Me.autoCompletePost)
            Me.Controls.Add(Me.post)
            Me.Controls.Add(Me.currentValue)
            Me.Controls.Add(Me.currentValueLabel)
            Me.Name = "TXForm"
            Me.Text = "TXForm"
            Me.ResumeLayout(False)
            Me.PerformLayout()

        End Sub 'InitializeComponent

#End Region

        Protected Overrides Sub OnLoad(ByVal e As EventArgs)
            Dim TXDemoServer As TXDemoServer = Nothing
            Try
                TXDemoServer = New TXDemoServer

                currentValue.Text = TXDemoServer.RetrieveCurrentValue().ToString(CultureInfo.CurrentCulture)
            Catch cv As CurrentValueNotReadException
                MessageBox.Show("Unable to read the current value from " & _
                  "the database", "Error")
                Application.Exit()
            Catch ex As Exception
                MessageBox.Show("An exception was caught : " & ex.Message & _
                 " Unable to launch the COM+ Server. Closing the " & _
                  "application.", "Error")
                Application.Exit()
            Finally
                'It is important to dispose COM+ objects as soon as possible so that
                'COM+ metadata such as context does not remain in memory unnecessarily
                'and so that COM+ services such as Object Pooling work properly.
                If Not TXDemoServer Is Nothing Then
                    TXDemoServer.Dispose()
                End If
            End Try
        End Sub

        'Start the Post with a manual transaction if the user clicks on the 
        'Post button
        Private Sub Post_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles post.Click

            Dim val As Integer = 0
            Try
                val = Integer.Parse(newValue.Text, CultureInfo.CurrentCulture)
            Catch
                MessageBox.Show("Please enter a value to post!")
            End Try

            Dim TXDemoServer As TXDemoServer = Nothing
            Try
                TXDemoServer = New TXDemoServer
                TXDemoServer.Post(val)
            Catch de As DatabaseExecutionException
                MessageBox.Show("Unable to update the database", "Error")
                Exit Sub
            Catch ex As Exception
                MessageBox.Show("An exception was caught : " + ex.Message)
                Exit Sub
            Finally
                'It is important to dispose COM+ objects as soon as possible so that
                'COM+ metadata such as context does not remain in memory unnecessarily
                'and so that COM+ services such as Object Pooling work properly.
                If Not TXDemoServer Is Nothing Then
                    TXDemoServer.Dispose()
                End If
            End Try
            Try
                TXDemoServer = New TXDemoServer
                currentValue.Text = TXDemoServer.RetrieveCurrentValue().ToString(CultureInfo.CurrentCulture)
            Catch cv As CurrentValueNotReadException
                MessageBox.Show("Unable to read the current value from the database", _
                 "Error")
                Exit Sub
            Finally
                'It is important to dispose COM+ objects as soon as possible so that
                'COM+ metadata such as context does not remain in memory unnecessarily
                'and so that COM+ services such as Object Pooling work properly.
                If Not TXDemoServer Is Nothing Then
                    TXDemoServer.Dispose()
                End If
            End Try
        End Sub


        'Start the Post with automatic transaction if the user clicks on the 
        'AutoCompletePost button
        Private Sub AutoComplete_PostClick(ByVal sender As Object, ByVal e As _
          System.EventArgs) Handles autoCompletePost.Click
            Dim val As Integer = 0
            Try
                val = Integer.Parse(newValue.Text, CultureInfo.CurrentCulture)
            Catch
                MessageBox.Show("Please enter a value to post!")
            End Try

            Dim TXDemoServer As TXDemoServer = Nothing
            Try
                TXDemoServer = New TXDemoServer
                TXDemoServer.AutoCompletePost(val)
            Catch at As ValueOutOfRangeException
                MessageBox.Show("The transaction has been aborted by " & _
                  "throwing an AbortTransactionException", "No Error")
            Catch db As DatabaseExecutionException
                MessageBox.Show("Unable to update the database", "Error")
                Exit Sub
            Catch ex As Exception
                MessageBox.Show("An exception was caught : " + ex.Message)
                Exit Sub
            Finally
                'It is important to dispose COM+ objects as soon as possible so that
                'COM+ metadata such as context does not remain in memory unnecessarily
                'and so that COM+ services such as Object Pooling work properly.
                If Not TXDemoServer Is Nothing Then
                    TXDemoServer.Dispose()
                End If
            End Try
            Try
                TXDemoServer = New TXDemoServer
                currentValue.Text = TXDemoServer.RetrieveCurrentValue().ToString(CultureInfo.CurrentCulture)
            Catch cv As CurrentValueNotReadException
                MessageBox.Show("Unable to read the current value from " & _
                 "the database", "Error")
                Exit Sub
            Finally
                'It is important to dispose COM+ objects as soon as possible so that
                'COM+ metadata such as context does not remain in memory unnecessarily
                'and so that COM+ services such as Object Pooling work properly.
                If Not TXDemoServer Is Nothing Then
                    TXDemoServer.Dispose()
                End If
            End Try
        End Sub

        Public Shared Sub Main(ByVal args() As String)
            Application.Run(New TXForm)
        End Sub

    End Class
End Namespace
